ckan/new_authz.py

```diff
@@ -431,12 +431,13 @@ def auth_is_anon_user(context):
 
         See ckan/lib/base.py:232 for pylons context object logic
     '''
```

Comment:
Do we support custom validators on user fields yet? Yes, I'm also terrified about mixing IP addresses and usernames, which is why I want to fix that, in a separate PR.

The tests broke because the new tests relied on getting a TypeError because no context had been set up yet. With the new test order an context now does exist (more reason to avoid using c) but is empty so `c.user == ''` and `c.author == ''` which made `is_anon_user == False`. This breaks the test.